### PR TITLE
Fix/delete app impersonate

### PR DESCRIPTION
**Pull request by gsps1**
Build running : http://builds.cask.co/browse/CDAP-DUT4547-1
https://issues.cask.co/browse/CDAP-6636

### CODE REVIEW

**Comment by gokulavasan** [comment]
LGTM 👍

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/deploy/pipeline/DeletedProgramHandlerStage.java [hunk 1]
```diff
@@ -101,21 +107,28 @@ public void process(ApplicationDeployable appSpec) throws Exception {
...
+        final String namespace = String.format("%s.%s", programId.getApplicationId(), programId.getId());
+
+        final NamespaceId namespaceId = appSpec.getApplicationId().getParent();
+        impersonator.doAs(namespaceId, new Callable<Void>() {
+
+          @Override
+          public Void call() throws Exception {
+            for (Map.Entry<String, Collection<Long>> entry : streamGroups.asMap().entrySet()) {
+              streamConsumerFactory.dropAll(namespaceId.stream(entry.getKey()).toId(), namespace, entry.getValue());
+              queueAdmin.dropAllForFlow(Id.Flow.from(programId.getApplication(), programId.getId()));
```
Comment (anwar6953) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/deploy/pipeline/DeletedProgramHandlerStage.java]: this should be outside the for-loop.

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/services/ApplicationLifecycleService.java [hunk 2]
```diff
@@ -583,21 +587,29 @@ private void deleteApp(Id.Application appId, ApplicationSpecification spec) thro
...
+      final Id.Flow flowId = Id.Flow.from(appId, flowSpecification.getName());
+      impersonator.doAs(new NamespaceId(appId.getNamespaceId()), new Callable<Void>() {
 
-      queueAdmin.dropAllForFlow(Id.Flow.from(appId, flowSpecification.getName()));
+        @Override
+        public Void call() throws Exception {
+          for (Map.Entry<String, Collection<Long>> entry : streamGroups.asMap().entrySet()) {
+            streamConsumerFactory.dropAll(Id.Stream.from(appId.getNamespaceId(), entry.getKey()),
+                                          namespace, entry.getValue());
+            queueAdmin.dropAllForFlow(flowId);
```
Comment (anwar6953) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/services/ApplicationLifecycleService.java]: same here

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/deploy/pipeline/DeletedProgramHandlerStage.java (hunk 1 above)
Comment (gsps1) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/deploy/pipeline/DeletedProgramHandlerStage.java]: oops. fixed it.

**Comment by anwar6953** [comment]
changes lgtm

**Pull request merged**